cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/remote/RemoteNamespaceQueryTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.store.remote;
+
+import co.cask.cdap.common.namespace.RemoteNamespaceQueryClient;
+import co.cask.cdap.gateway.handlers.meta.RemoteNamespaceQueryHandler;
+import co.cask.cdap.internal.app.services.http.AppFabricTestBase;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.NamespaceConfig;
+import co.cask.cdap.proto.NamespaceMeta;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.store.NamespaceStore;
+import com.google.inject.Injector;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+
+/**
+ * Tests implementation of {@link RemoteNamespaceQueryHandler} by using it fetch namespaces.
+ */
+public class RemoteNamespaceQueryTest extends AppFabricTestBase {
+
+  private static NamespaceStore namespaceStore;
+  private static RemoteNamespaceQueryClient queryClient;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    Injector injector = getInjector();
+    namespaceStore = injector.getInstance(NamespaceStore.class);
+    queryClient = injector.getInstance(RemoteNamespaceQueryClient.class);
+  }
+
+  @Test
+  public void testDefaultNS() throws Exception {
+    List<NamespaceMeta> nsMetaList = queryClient.list();
+    boolean foundDefault = false;
+    for (NamespaceMeta meta : nsMetaList) {
+      foundDefault = meta.getName().equals(Id.Namespace.DEFAULT.getId());
+      if (foundDefault) {
+        break;
+      }
+    }
+    Assert.assertTrue(foundDefault);
+  }
+
+  @Test
+  public void testCustomNS() throws Exception {
+    String cdapNamespace = "NS1";
+    String hbaseNamespace = "custHBase";
+    String rootDirectory = "/directory";
+    String hiveDb = "myHive";
+    String schedulerQueue = "schQ";
+    String description = "Namespace with custom HBase mapping";
+    NamespaceConfig namespaceConfig = new NamespaceConfig(schedulerQueue, rootDirectory, hbaseNamespace, hiveDb);
+    namespaceStore.create(new NamespaceMeta.Builder().setName(cdapNamespace).setDescription(description)
```

Comment:
nit: just for readability, can you put each method call in this builder on its own line?